src/avt/Pipeline/Data/avtDataRepresentation.C

```diff
@@ -408,10 +410,33 @@ ConvertVTKToVTKm(vtkDataSet *data)
 
             ds.AddCoordinateSystem(
                 vtkm::cont::CoordinateSystem("coordinates", coordinates));
+#else
+            // The way Ascent sets the coordinates.
+            float *pts = static_cast<float*>(points->GetVoidPointer(0));
+            vtkm::cont::ArrayHandle<vtkm::Float32> x_coords;
+            vtkm::cont::ArrayHandle<vtkm::Float32> y_coords;
+            vtkm::cont::ArrayHandle<vtkm::Float32> z_coords;
+            x_coords.Allocate(nPoints);
+            y_coords.Allocate(nPoints);
+            z_coords.Allocate(nPoints);
+
+            for (vtkm::Id i = 0; i < nPoints; ++i)
+            {
+                x_coords.WritePortal().Set(i, pts[i*3]);
+                y_coords.WritePortal().Set(i, pts[i*3+1]);
+                z_coords.WritePortal().Set(i, pts[i*3+2]);
+            }
+
+            ds.AddCoordinateSystem(
+                vtkm::cont::CoordinateSystem("coordinates",
+                    make_ArrayHandleSOA(x_coords, y_coords, z_coords)));
+#endif
         }
         else if (points->GetDataType() == VTK_DOUBLE)
         {
-            double *pts = static_cast<double*>(points->GetVoidPointer(0));
+#if 0
```

Comment:
Does this need to remain?

Reply:
I switched to using the Ascent way to define the coordinates thinking that may be the source of the crash when processing using the VTK-m marching cells filter with unstructured meshes, but it didn't fix it. I would like to have @cyrush weigh in on which he thinks is better. It seems like a slower conversion, but maybe it operates faster on the GPU.

Reply:
Sound good, thanks.